### DISCUSSION TITLE
TaxCloud support (USA sales tax)

### DISCUSSION THREAD

**Discussion by onecrayon** [saleor/saleor#14435] [Ideas]
### What I'm trying to achieve

As far as we can tell, the existing tax plug-ins in Saleor are all supporting the collection of taxes in Europe (USA-based taxes are based on the originating shipment location vs. the eventual delivery location; a country-based approach makes no sense in this context).

### Describe a proposed solution

Currently, we are planning to implement a [TaxCloud](https://taxcloud.com/home) plugin for internal use that will lookup taxes via TaxCloud just-in-time (ignoring the core cacheing functionality), but we wanted to check if there were any prior art, community interest beyond our company, or plans on Mirumee's part to natively support USA-based sales tax situations.

**Comment by patrys** [comment]
Saleor ships with an Avalara Avatax plugin that does support US taxes. If you still want to use TaxCloud, please consider modeling the plugin after the Avatax one (and contributing it 😄).

**Comment by onecrayon** [comment]
@patrys Interesting; we'll take a look. Odds are good we'll stick with TaxCloud simply because we are already using them for other projects, but if we do implement at TaxCloud plug-in we'll certainly contribute it.

**Comment by patrys** [comment]
@onecrayon We are currently designing a synchronous webhook callback for taxes (taking checkout/order data as input and returning tax information for lines, shipping, and totals). I believe it would make implementing TaxCloud as a Saleor App (as opposed to a plugin) not only possible but also easier than implementing it as a plugin using the current API. I'm not sure about your timeline but I'd expect it to land in Saleor in about a month.

**Reply by pljspahn** [reply]
Curious if this got implemented.

I'm brainstorming a marketplace style shop and I want to avoid any paid tax provider (Avalara, TaxJar, etc) whenever possible. I would need a pretty in-depth custom sales tax app that can be scoped for each vendor.

**Reply by korycins** [reply]
> Curious if this got implemented.

https://docs.saleor.io/developer/taxes#dynamic-taxes

**Comment by onecrayon** [comment]
@patrys Awesome, that's really good to hear. We'll keep that in mind; we haven't yet started implementing TaxCloud support, but that definitely sounds like a superior solution (since the current method hooks would potentially result in a lot more API calls when we really only need one per checkout/order).
